ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/damle/SandboxDamle.scala

```diff
@@ -19,12 +21,15 @@ import com.digitalasset.daml.lf.value.Value.AbsoluteContractId
 import com.digitalasset.daml.lf.language.Ast.Package
 import com.digitalasset.daml.lf.data.Ref
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ExecutionContext, Future, Promise}
 
 /**
   * provides smart constructor for DAML engine environment
   */
 object SandboxDamle {
+  // Concurrent map of promises to request each package only once.
+  private[this] val packagePromises: ConcurrentHashMap[Ref.PackageId, Promise[Option[Package]]] =
```

Comment:
```suggestion
  private val packagePromises: ConcurrentHashMap[Ref.PackageId, Promise[Option[Package]]] =
```
Generally a good practice but not needed for an `object`.